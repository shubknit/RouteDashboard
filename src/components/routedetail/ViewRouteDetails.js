import React from 'react';

// This component is display a route details or any error if found during API response
export const ViewRouteDetails = (props) => (
	<div className ="route-info">
		{ props.data.isFetching && <p>Loading...</p> }
        { props.data.mapData.distance && <p> total distance: { props.data.mapData.distance } </p> }
        { props.data.mapData.time && <p> total time: { props.data.mapData.time } </p> }
        { props.data.error && <p className = 'error'>  { props.data.error } </p> } 
	</div>
)