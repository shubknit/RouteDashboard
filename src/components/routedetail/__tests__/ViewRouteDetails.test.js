import React from 'react';
import { ViewRouteDetails } from '../ViewRouteDetails';
import renderer from 'react-test-renderer';


const state = {
    mapData:  {
        distance: '',
         time: ''
    },
    isFetching: '',
    error: '',
}

it('should render the ViewRouteDetails correctly', () => {
    const wrapper = renderer.create(<ViewRouteDetails data = { state }/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})