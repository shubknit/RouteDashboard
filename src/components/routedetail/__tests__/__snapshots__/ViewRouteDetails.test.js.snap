// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ViewRouteDetails correctly 1`] = `
<div
  className="route-info"
>
  
  
  
  
</div>
`;
